src/objects/basic_npc.py

```diff
@@ -111,17 +92,11 @@ def take_damage(self, damage):
         if self.hp <= 0:
             self.alive = False
 
-    def shoot_bullet(self, collision_sensor, bullet_id):
+    def shoot_bullet(self, offset, collision_sensor):
         self.bullet_ready = False
         self.prev_shot_time = time.time()
-        # offset: radius of the shooter to avoid bullet collision with the shooter itself
-        offset = self.radius + self.bullet_radius
-        bullet_pos = self.pos + offset * self.dir
-        self.game_stats.track_bullet_fired()
-        return Bullet(bullet_pos, self.dir, self.damage, self.bullet_radius, collision_sensor, bullet_id)
 
-    def pick_up(self, item):
-        self.items.append(item)
```

Comment:
Again, check for the latest status of the branch, before adding files. This method was probably added after your last fetch, so if you commit over it with the same filename these methods are lost in this file, however they will be called in other files, as you did not modify them. This breaks the code structure, and it won't be able to run. Ideally, one shouldn't use file uploads when modifying already existing files in the repository, rather fetch->modify->commit changes.